modules/swagger-codegen/src/main/resources/python/api_client.mustache

```diff
@@ -625,3 +640,124 @@ class ApiClient(object):
+        :param data: Data set by User
+        :return: instance of digest object 
+        """
+        digest = SHA256.new()
+        digest.update(data)
+        
+        return digest
+    
+    def get_rsasig_b64encode(self, private_key_path, digest):
+        """
+        :param private_key_path : abs path to private key .pem file.
+        :param digest: digest
+        :return: instance of digest object 
+        """
+        
+        key = open(private_key_path, "r").read()
+        rsakey = RSA.importKey(key)
+        signer = PKCS1_v1_5.new(rsakey)
+        sign = signer.sign(digest)
+        
+        return b64encode(sign)
+    
+    def prepare_str_to_sign(self, req_tgt, hdrs):
+        """
+        :param req_tgt : Request Target as stored in http header.
+        :param hdrs: HTTP Headers to be signed.
+        :return: instance of digest object 
+        """
+        ss = ""
+        ss = ss + "(request-target): " + req_tgt.lower() + "\n"
+    
+        length = len(hdrs.items())
+    
+        i = 0
+        for key, value in hdrs.items():
+            ss = ss + key.lower() + ": " + value
+            if i < length-1:
+                ss = ss + "\n"
+            i += 1
+        
+        return ss
+
+    def get_auth_header(self, auth_settings, hdrs, signed_msg):
+        """
+        This method prepares the Auth header string
+        
+        :param hdrs : HTTP Headers
+        :param signed_msg: Signed Digest
+        :return: instance of digest object 
+        """
+        
+        auth_str = ""
+        auth_str = auth_str + "Signature"
+    
+        auth_str = auth_str + " " + "keyId=\"" + auth_settings['api_key_id'] + "\"," + "algorithm=\"" + auth_settings['digest_algorithm'] + "\"," + "headers=\"(request-target)"
+    
+        for key, _ in hdrs.items():
+            auth_str = auth_str + " " + key.lower()
+        auth_str = auth_str + "\""
+    
+        auth_str = auth_str + "," + "signature=\"" + signed_msg.decode('ascii') + "\""
+    
+        return auth_str
+    
+    def prepare_auth_header(self, auth_settings, resource_path, method, body, query_params):
+        """
+        Method to perform operations required to prepare the auth header and eventually
+        add the signed headers to default_header object used by ApiClient class.
+        
+        https://datatracker.ietf.org/doc/draft-cavage-http-signatures/?include_text=1
+        
+        :param auth_setting: Authsettings dictionary
+        :param resource_path : resource path which is the api being called upon.
+        :param method: request type
+        :param body: body passed in the http request.
+        :param query_params: query parameters used by the API 
+        :return: instance of digest object 
+        """
+        if body is None:
+            body = ''
+        else:
+            body = json.dumps(body)
+        
+        
+        target_host = urlparse(self.host).netloc
+        target_path = urlparse(self.host).path
```

Comment:
There is no reason to parse host twice and I don't expect host to contain netloc and path at all.